pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -22,12 +25,14 @@ import (
 	k8sappsv1 "k8s.io/api/apps/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
 )
 
 const (
-	repoName          = "rancher-charts"
-	sucDeploymentName = "system-upgrade-controller"
-	legacyAppLabel    = "io.cattle.field/appId"
+	repoName           = "rancher-charts"
+	sucDeploymentName  = "system-upgrade-controller"
+	legacyAppLabel     = "io.cattle.field/appId"
+	legacyAppFinalizer = "systemcharts.cattle.io/legacy-k3s-based-upgrader-deprecation"
```

Comment:
Is `systemcharts.cattle.io` established name or something we're inventing specifically for this?